sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -354,9 +355,14 @@ resource
     : identifier STRING
     ;
 
+insertStatement
+    : (ctes)? insertInto queryTerm queryOrganization                                       #singleInsertQuery
```

Comment:
Can you move `(ctes)? ` to the line 80 like `| (ctes)? insertStatement   `?

Reply:
@maropu then i may have to implement/override another visit method to apply the cte to an insert statement, okay ?

Reply:
the duplicate patten?
```
queryToDesc
    : queryTerm queryOrganization
    ;
```

Reply:
@maropu Thats right. After this PR is in, i will have a follow-up to allow CTEs for DESCRIBE QUERY and that would remove this pattern all together.